utils/weight-diff/src/main.rs

```diff
@@ -0,0 +1,261 @@
+        #[arg(ignore_case = true, value_enum)]
+        runtime: Runtime,
+        /// for which weights to generate a table?
+        #[arg(ignore_case = true, value_enum)]
+        kind: WeightsKind,
+        /// if present, displays the value in units
+        #[arg(long)]
+        display_units: bool,
+    },
+}
+
+#[derive(Debug, Clone, ValueEnum)]
+enum Runtime {
+    Gear,
+    Vara,
+}
+
+#[derive(Debug, Clone, ValueEnum)]
+enum WeightsKind {
+    Instruction,
+    HostFn,
+}
+
+#[derive(Debug, Serialize)]
+struct SerializableDump {
+    gear_schedule: Schedule<gear_runtime::Runtime>,
+    vara_schedule: Schedule<vara_runtime::Runtime>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    label: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct DeserializableDump {
+    gear_schedule: DeserializableSchedule,
+    vara_schedule: DeserializableSchedule,
+    label: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct DeserializableSchedule {
+    instruction_weights: IndexMap<String, serde_json::Value>,
+    host_fn_weights: IndexMap<String, serde_json::Value>,
+}
+
+impl DeserializableSchedule {
+    fn instruction_weights(&self) -> IndexMap<String, u64> {
+        let mut map = IndexMap::new();
+
+        for (k, v) in self.instruction_weights.clone() {
+            if k == "version" {
+                continue;
+            }
+
+            if let Ok(v) = serde_json::from_value(v) {
+                map.insert(k, v);
+            }
+        }
+
+        map
+    }
+
+    fn host_fn_weights(&self) -> IndexMap<String, u64> {
+        let mut map = IndexMap::new();
+
+        for (k, v) in self.host_fn_weights.clone() {
+            if let Ok(v) = serde_json::from_value::<Weight>(v) {
+                map.insert(k, v.ref_time());
+            }
+        }
+
+        map
+    }
+}
+
+fn format_weight(weight: u64) -> String {
+    if weight > 1_000_000_000 {
+        format!(
+            "{:.1?} ms",
+            Fixed::saturating_from_rational(weight, 1_000_000_000).to_float(),
+        )
+    } else if weight > 1_000_000 {
+        format!(
+            "{:.1?} µs",
+            Fixed::saturating_from_rational(weight, 1_000_000).to_float(),
+        )
+    } else if weight > 1_000 {
+        format!(
+            "{:.1?} ns",
+            Fixed::saturating_from_rational(weight, 1_000).to_float(),
+        )
+    } else {
+        format!("{} ps", weight)
+    }
+}
+
+fn format_value(value: Option<u64>, display_units: bool) -> String {
+    value
+        .map(|v| {
+            if display_units {
+                format_weight(v)
+            } else {
+                format!("{v}")
+            }
+        })
+        .unwrap_or_else(|| "N/A".into())
+}
+
+fn format_diff(value1: Option<u64>, value2: Option<u64>) -> String {
+    value1
+        .filter(|&a| a != 0)
+        .zip(value2)
+        .map(|(a, b)| {
+            let (a, b) = (a as f64, b as f64);
+            let percentage_diff = ((b - a) / a) * 100.0;
```

Comment:
More common way in perf comparison tools is to calculate as
```
(old / new) -  1
```
It is also common that first param is old and second is a new one.

So in your case is 
```rust
let percentage_diff = (a / b -  1) * 100.0;
```
But it would be great if you rename `a` and `b` like in other places to `1` and `2`.